### ISSUE TITLE
Transform causes worker to crash

### ISSUE THREAD

**Issue by EliSnow** [pipelinedb/pipelinedb#1424]
The following causes one of the worker processes to crash which results in a server restart:

```sql
CREATE STREAM a_stream (n integer);
CREATE STREAM b_stream (n integer);

CREATE CONTINUOUS TRANSFORM repeat_transform AS
  SELECT n
  FROM a_stream
  CROSS JOIN generate_series(1, 10) AS gs
  THEN EXECUTE PROCEDURE pipeline_stream_insert('b_stream');

CREATE CONTINUOUS VIEW cv AS
SELECT
  n
FROM b_stream;

insert into a_stream (n)
  select n
  from generate_series(1, 1000000) AS gs (n);
```

There's not a whole lot of useful information in the logs:

```
LOG:  worker process: worker2 [stats] (PID 516) was terminated by signal 9: Killed
DETAIL:  Failed process was running: worker2 [stats]
LOG:  terminating any other active server processes
WARNING:  terminating connection because of crash of another server process
DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
```

It seems related to the `CROSS JOIN`. In my actual use case I have a transform which does a cross join using `jsonb_array_elements` to "unnest" the json coming through the stream.

**Comment by usmanm** [comment]
Non-inner stream-table joins are currently broken (#1386). Does this issue persist if you use an inner join?

**Comment by EliSnow** [comment]
I just tried with:

```sql
CREATE CONTINUOUS TRANSFORM repeat_transform AS
  SELECT n
  FROM a_stream
  INNER JOIN generate_series(1, 10) AS gs ON (true)
  THEN EXECUTE PROCEDURE pipeline_stream_insert('b_stream');
```

And still experienced the crash.

**Comment by EliSnow** [comment]
Thinking it was related to the implicit lateral join, I tried:

```js
CREATE CONTINUOUS TRANSFORM repeat_transform AS
  SELECT n
  FROM a_stream
  INNER JOIN (SELECT m FROM generate_series(1, 10) AS gs (m)) x ON (true)
  THEN EXECUTE PROCEDURE pipeline_stream_insert('b_stream');
```

But the crash still persists, so this is not related to lateral joins.

**Comment by usmanm** [comment]
It probably is a memory leak which is causing the process to get killed.

**Comment by usmanm** [comment]
This should be fixed by #1423, going to close for now.
